Library/Homebrew/cmd/test-bot.rb

```diff
@@ -651,7 +651,8 @@ def cleanup_before
         safe_system "git", "checkout", "-f", "master"
         # This will uninstall all formulae, as long as
         # HOMEBREW_REPOSITORY == HOMEBREW_PREFIX, which is true on the test bots
-        safe_system "git", "clean", "-ffdx", "--exclude=/Library/Taps/" unless ENV["HOMEBREW_RUBY"] == "1.8.7"
+        Tap.names.reject { |t| t == "homebrew/core" }.each { |s| safe_system "brew untap #{s}" }
```

Comment:
If I'm reading the code of `test-bot` correctly, `before_cleanup` will be executed after [installing the tap required for the run](https://github.com/Homebrew/brew/blob/f15b631875ef19599bcb4e774d3e67596d1f93ac/Library/Homebrew/cmd/test-bot.rb#L894-L909). Thus, running `test-bot`, e.g., for `homebrew/versions` will install this tap and then immediately untap it again. Right?

Just a detail, but I think we prefer using `safe_system` with separate arguments to avoid spawning an intermediate shell and to prevent undesired interpretation of arguments by that shell (unlikely in this particular case).

Finally, the chained `reject` and `each` can be simplified a little bit by writing:

```ruby
Tap.names.each { |t| safe_system "brew", "untap", s if s != "homebrew/core" }
```